src/theme-provider.js

```diff
@@ -16,6 +19,16 @@ class ThemeProvider extends Component {
   // create theme, by merging with outer theme, if present
   getTheme(passedTheme) {
     const theme = passedTheme || this.props.theme
+    if (isFunction(theme)) {
+      const mergedTheme = theme(this.outerTheme)
+      if (!isPlainObject(mergedTheme)) {
+        throw new Error(
```

Comment:
Do `PropType` validation failures lead to thrown exception in production mode in `React`? I can't recall off the top of my head.

If a `PropType.object` validation would only _warn_, we might want to be more liberal here and do the same. But this needs clarification, as I'm unsure myself